server/src/main/java/org/elasticsearch/Version.java

```diff
@@ -268,6 +269,15 @@ private static Version fromStringSlow(String version) {
         this.build = (byte) (id % 100);
         this.luceneVersion = Objects.requireNonNull(luceneVersion);
         this.toString = major + "." + minor + "." + revision;
+        this.previousMajorId = major > 0 ? (major - 1) * 1000000 + 99 : major;
+    }
+
+    public Version previousMajor() {
+        return Version.fromId(previousMajorId);
+    }
+
+    public static Version minimumRestCompatibilityVersion() {
```

Comment:
hmm, should we make this non-static like `minimumCompatibilityVersion` and `minimumIndexCompatibilityVersion`? Also, please add javadocs :)